firmware/hw_layer/rtc_helper.cpp

```diff
@@ -4,73 +4,92 @@
-	rtcConvertDateTimeToStructTm(&timespec, timp, NULL);
-#endif /* EFI_RTC */
-}
+#include "rusefi_types.h"
+#endif // EFI_RTC
 
 #if EFI_PROD_CODE
+#include <sys/time.h>
+
 // Lua needs this function, but we don't necessarily have to implement it
 extern "C" int _gettimeofday(timeval* tv, void* tzvp) {
 	(void)tv; (void)tzvp;
 	return 0;
 }
-#endif
+#endif // EFI_PROD_CODE
 
 #if EFI_RTC
-static time_t GetTimeUnixSec() {
-	tm tim;
-	RTCDateTime timespec;
+void initRtc() {
+	efiPrintf("initRtc()");
+	printDateTime(); // this would test RTC, see 'rtcWorks' variable, see #311
+}
 
-	rtcGetTime(&RTCD1, &timespec);
-	rtcConvertDateTimeToStructTm(&timespec, &tim, NULL);
-	time_t result = mktime(&tim);
+static const char * const monthAbbrs[] = {
+	"Jan", "Feb", "Mar",
+	"Apr", "May", "Jun",
+	"Jul", "Aug", "Sep",
+	"Oct", "Nov", "Dec"
+};
+
+void printRtcDateTime() {
+	efidatetime_t dateTime = getRtcDateTime();
+	// prints the date like: 19 sep 2022 21:19:55
+	efiPrintf("Current RTC time: %02u %s %04u %02u:%02u:%02u",
+			dateTime.day, monthAbbrs[dateTime.month - 1], dateTime.year,
+			dateTime.hour, dateTime.minute, dateTime.second);
+}
 
-	return result;
+void setRtcDateTime(efidatetime_t const * const dateTime) {
+	RTCDateTime timespec = convertRtcDateTimeFromEfi(dateTime);
+	rtcSetTime(&RTCD1, &timespec);
 }
 
-static void SetTimeUnixSec(time_t unix_time) {
-	tm tim;
+efidatetime_t getRtcDateTime() {
+	RTCDateTime timespec;
+	rtcGetTime(&RTCD1, &timespec);
+	return convertRtcDateTimeToEfi(&timespec);
+}
 
-#if defined __GNUC__
-	tm *canary;
-	/* If the conversion is successful the function returns a pointer
-		to the object the result was written into.*/
-	canary = localtime_r(&unix_time, &tim);
-	osalDbgCheck(&tim == canary);
-#else
-	tm *t = localtime(&unix_time);
-	memcpy(&tim, t, sizeof(tm));
-#endif
+efidatetime_t convertRtcDateTimeToEfi(RTCDateTime const * const timespec) {
+	uint32_t second = timespec->millisecond / 1000;
+	uint16_t minute = second / 60;
+	second -= minute * 60;
+	uint8_t hour = minute / 60;
+	minute -= hour * 60;
+
+	efidatetime_t const dateTime = {
+		.year = timespec->year + RTC_BASE_YEAR,
+		.month = (uint8_t)timespec->month,
+		.day = (uint8_t)timespec->day,
+		.hour = hour,
+		.minute = (uint8_t)minute,
+		.second = (uint8_t)second,
+	};
+	return dateTime;
+}
 
+RTCDateTime convertRtcDateTimeFromEfi(efidatetime_t const * const dateTime) {
 	RTCDateTime timespec;
-	rtcConvertStructTmToDateTime(&tim, 0, &timespec);
-	rtcSetTime(&RTCD1, &timespec);
+	timespec.year = dateTime->year - RTC_BASE_YEAR; // ChibiOS year origin is e.g. 1980
+	timespec.month = dateTime->month; // [1..12]
+	timespec.day = dateTime->day; // [1..31]
+	timespec.millisecond = (((dateTime->hour * 60) + dateTime->minute) * 60 + dateTime->second) * 1000; // ms since midnight
+	timespec.dayofweek = RTC_DAY_CATURDAY; // CATURDAY: 0 ... ?
+	timespec.dstflag = 0; // 0 ... ?
+	return timespec;
 }
 
+// TODO(nms): move to e.g. efitime ?
```

Comment:
probably just drop this comment? these methods are RTC dependent, but not actually used close to ChibiOS RTC like the rest